dss-builder/rootfs.skel/usr/bin/dss-deploy-pilot

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+"""
+Service deploy agent.
+
+This agent is installed on the builder instance, which is configured and launched by scripts/launch_dss_builder.sh.
+
+When the instance starts, the agent is launched by the systemd service configured by
+dss-builder/rootfs.skel/etc/systemd/system/multi-user.target.wants/dss-deploy-pilot-dev.service
+(/etc/systemd/system/multi-user.target.wants/dss-deploy-pilot-dev.service on the instance).
+
+The agent polls GitHub for new deployment objects, which are created by running "make .
+"""
+
+import os, sys, subprocess, json, logging, argparse, shutil, signal, socket, time
+import boto3, requests
+from datetime import datetime
+
+def get_metadata(path):
+    return requests.get("http://169.254.169.254/latest/meta-data/{}".format(path)).content.decode()
+
+def clean_old_builds(build_root, prefix, min_old_builds=2):
+    build_dirs = [os.path.join(build_root, d) for d in os.listdir(build_root) if d.startswith(prefix)]
+    build_dirs = [d for d in build_dirs if os.path.isdir(d) and not os.path.islink(d)]
+    build_dirs = sorted(build_dirs, key=lambda d: os.stat(d).st_mtime)
+    for d in build_dirs[:-min_old_builds]:
+        logging.warn("Deleting old build %s", d)
+        shutil.rmtree(d, ignore_errors=True)
+
+def run_git_command(*cmd, **kwargs):
+    kwargs["env"] = dict(kwargs.get("env", os.environ), GIT_SSH_COMMAND="aegea-git-ssh-helper")
+    return subprocess.check_call(["git"] + list(cmd), **kwargs)
+
+def clone_and_build(branch, destination):
+    if not os.path.exists(os.path.dirname(destination)):
+        os.makedirs(os.path.dirname(destination))
+    deploy_dir = "{}-{}-{}".format(args.gh_repo_name, branch, datetime.now().strftime("%Y-%m-%d-%H-%M-%S"))
```

Comment:
using args as a global?

Reply:
Yes - something wrong with that? This is a monolithic script.

Reply:
Fair enough, that's fine